client/v3/retry_interceptor.go

```diff
@@ -74,6 +74,12 @@ func (c *Client) unaryClientInterceptor(logger *zap.Logger, optFuncs ...retryOpt
 				continue
 			}
 			if callOpts.retryAuth && rpctypes.Error(lastErr) == rpctypes.ErrInvalidAuthToken {
+				// clear auth token before refreshing it.
```

Comment:
@agargi why are you doing this? This doesn't seem right. You might want to rebase your code. Thanks!

Reply:
@spzala Apologies for missing rebase. Should have taken care of it earlier. Fixed now

Reply:
No problem at all @agargi  Thanks for fixing it.